sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -1625,6 +1625,15 @@ object SQLConf {
     .intConf
     .createWithDefault(25)
 
+  val MAX_PLAN_STRING_LENGTH = buildConf("spark.sql.maxPlanLength")
+    .doc("Maximum number of characters to output for a plan string.  If the plan is " +
```

Comment:
Do you want to restrict only one plan or all dumped plans?

Reply:
The intent is all dumped plans.  Based on the comment by @felixcheung above we set the default to not do any truncation, so the default behavior is that same as the existing behavior.